curriculum/challenges/english/14-responsive-web-design-22/learn-css-flexbox-by-building-a-photo-gallery/6493bc0d99879635209565aa.md

```diff
@@ -0,0 +1,78 @@
+---
+id: 6493bc0d99879635209565aa
+title: Step 8
+challengeType: 0
+dashedName: step-8
+---
+
+# --description--
+
+Ooh you have a problem. See how the images overflow the container? That's because the `box-sizing property` is at the `content-box` value, aka the default. This doesn't include the padding and border when calculating an element's size. Now, to solve this issue, use `border-box`, the box-sizing property's second value. Unlike `content-box`, `border-box` excludes the margins and only adds the padding and border in calculating the size of the element. Try changing the box-sizing value from one to the other, and see the difference.
+Lastly, give the img elements a uniform height of 300px.
+
+
+# --hints--
+
+
+Your * selector should have a `box-sizing` property set to `border-box` as the value.
+
+```js
+assert.equal(new __helpers.CSSHelp(document).getStyle('*')?.boxSizing, 'border-box');
+```
+
+
+# --seed--
+
+## --seed-contents--
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Photo Gallery</title>
+    <link rel="stylesheet" href="./styles.css">
+  </head>
+  <body>
+    <header class="header">
+      <h1>css flexbox photo gallery</h1>
+    </header>
+    <div class="gallery">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/1.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/2.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/3.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/4.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/5.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/6.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/7.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/8.jpg">
+      <img src="https://cdn.freecodecamp.org/curriculum/css-photo-gallery/9.jpg">
+    </div>
+  </body>
+</html>
+
+
+```
+
+```css
+.gallery {
+  border: 2px solid red;
+  width: 50%;
+  height: 50%;
+}
+
+img {
+  width: 100%;
+  height: 100%;
+  padding: .5em;
+  border: 1px solid green;
+  box-sizing: content-box;
+}
```

Comment:
None of this CSS is present in the previous step. We should never automatically add code for campers, unless there is a specific reason and we are very explicit about it.

Reply:
If it is not necessary for the step, yes.

If we need this CSS to demonstrate the box model effectively, we should add steps where the camper adds it.

Reply:
Oh I see. Can you help me remove it? I’m not near my computer at the moment and it’s hard to do it on the GH mobile app.

Reply:
Sure. I'll dig into this.